LocalMultiplayerAgent/DeploymentTool/DeploymentScript.cs

```diff
@@ -0,0 +1,256 @@
+        private readonly DeploymentSettings settingsDeployment;
+        public DeploymentScript(MultiplayerSettings multiplayerSettings)
+        {
+            settings = multiplayerSettings;
+            settingsDeployment = JsonConvert.DeserializeObject<DeploymentSettings>(File.ReadAllText("DeploymentTool/deployment.json"));
+        }
+
+        public async Task RunScriptAsync()
+        { 
+            PlayFabSettings.staticSettings.TitleId = settings.TitleId;
+
+            string secret = Environment.GetEnvironmentVariable("PF_SECRET");
+            if (string.IsNullOrEmpty(secret))
+            {
+                Console.WriteLine("Enter developer secret key");
+                PlayFabSettings.staticSettings.DeveloperSecretKey = Console.ReadLine();
+            }
+
+            var req = new PlayFab.AuthenticationModels.GetEntityTokenRequest();
+            var res = await PlayFabAuthenticationAPI.GetEntityTokenAsync(req);
+
+            if (res.Error != null)
+            {
+                Console.WriteLine(res.Error.ErrorMessage);
+                return;
+            }
+
+            // TODO: do validation checks
+
+            dynamic createBuild = null;
+            if (settings.RunContainer)
+            {
+                if (settingsDeployment.OSPlatform == "Windows")
+                {
+                    CreateBuildWithManagedContainerRequest request = GetManagedContainerRequest();
+
+                    foreach (var file in request.GameAssetReferences)
+                    {
+                        await CheckAssetFiles(file.FileName);
+                    }
+
+                    createBuild = await CreateBuildWithManagedContainer(request);
+                }
+                else
+                {
+                    //for CustomLinux Container
+                    //yet to be implemented
+                }
+            }
+            else
+            {
+                CreateBuildWithProcessBasedServerRequest request = GetProcessBasedServerRequest();
+
+                foreach (var file in request.GameAssetReferences)
+                {
+                    await CheckAssetFiles(file.FileName);
+                }
+
+                createBuild = await CreateBuildWithProcessBasedServer(request);
+            }
+
+            if (createBuild.Error != null)
+            {
+                foreach (var error in createBuild.Error.ErrorDetails)
+                {
+                    foreach (var errorMessage in error.Value)
+                    {
+                        Console.WriteLine($"{errorMessage}");
+                    }
+                }
+            }
+        }
+
+        public CreateBuildWithCustomContainerRequest GetCustomContainerRequest()
+        {
+            return new CreateBuildWithCustomContainerRequest
+            {
+                BuildName = settingsDeployment.BuildName,
+                VmSize = (AzureVmSize)Enum.Parse(typeof(AzureVmSize), settingsDeployment.VmSize),
```

Comment:
If you do VmSize = Enum.Parse<AzureVmSize>(settingsDeployment.VmSize) here you can avoid the cast entirely and make it a strongly typed call.

Reply:
Sorry, it filtered out the lt and gt tags. It should be:

`VmSize = Enum.Parse<AzureVmSize>(settingsDeployment.VmSize)`

Let's see if that works